protocol.md

```diff
@@ -265,9 +263,9 @@ key SHOULD be ASCII encoded and the value MUST be Base64 encoded.
 ##### POST
 
 Clients MUST use a `POST` against a well known upload creation URL to request the
-creation of a new file resource. The request MUST include an `Upload-Length`
-header. If the [Stream](#stream) extension is used to upload a file of unknown
-size the header `Upload-Length: streaming` MUST be included.
+creation of a new file resource. The request MUST include `Upload-Length`
+header when upload size is known. The request MUST include `Upload-Stream: 1` when
+upload size is unknown. The headers included here SHALL be dictated by dependent extensions.
```

Comment:
Either move the mention of `Upload-Stream` to the Stream extension or link to it.